packages/nextjs/pages/_app.tsx

```diff
@@ -17,17 +18,27 @@ import "~~/styles/globals.css";
 const ScaffoldEthApp = ({ Component, pageProps }: AppProps) => {
   const price = useEthPrice();
   const setEthPrice = useAppStore(state => state.setEthPrice);
+  const [isDarkTheme, setIsDarkTheme] = useState(true);
+  const { isDarkMode } = useDarkMode();
 
   useEffect(() => {
     if (price > 0) {
       setEthPrice(price);
     }
   }, [setEthPrice, price]);
 
+  useEffect(() => {
+    setIsDarkTheme(isDarkMode);
```

Comment:
Why do you need the isDarkTheme state var? Can you use isDarkMode directly?

Reply:
We can't.
Initially even I approached it the same way by directly using the isDarkMode variable but seems that the value for isDarkMode variable defaults to false during server side rendering and it does get updated when it reaches the client side but the change is not seen unless useEffect and another variable is used to then re render on the client side on the initial load. Obviously everything works fine after the website is mounted and the theme is changed.

You can re produce this and check for yourself by, setting your OS theme to be darkmode and refresh the website. The website will load in dark theme but the rainbowkit model will still be in light mode unless you switch the theme to light and then dark mode again.

Reply:
Ok! Please add a small comment in the code explaining it, so we know why it's needed.

Reply:
Done

Reply:
Thanks!!